client/src/components/YelpRequestForm.js

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from "react"
+import { Redirect } from "react-router-dom"
+
+
+const YelpRequestForm = (props) => {
+  const [yelpParams, setYelpParams] = useState({})
+  const [shouldRedirect, setShouldRedirect] = useState(false)
+  const [requestParams, setRequestParams] = useState({
+    zip: "",
+    category: "",
+    term: "",
+    price: ""
+  })
+  
+  const queryYelp = async (requestPayload) => {
+    
+    try {
+      const response = await fetch("/api/v1/yelp", {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(requestPayload)
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json()
+          const newErrors = translateServerErrors(body.errors)
+          return setErrors(newErrors)
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`
+          const error = new Error(errorMessage)
+          throw error
+        } 
+      } else {
+        const body = await response.json()
+        setYelpParams(body)
+        console.log("BODY")
+        console.log(body)
+        
+        // yelpQueryResults = body.yelpQueryResults
+        setShouldRedirect(true)
+        
+      }
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+    
+    }
+    if (shouldRedirect) {
+      console.log(shouldRedirect)
+      console.log("YELP PARAMS")
+      console.log(yelpParams)
+      return(<Redirect 
+        to={{
+          pathname: "/places",
+          state: { places: yelpParams } 
```

Comment:
state should not be passed through a redirect where possible, we can instead redirect then through a useEffect fetch for the info that component needs